roxctl/generate/netpol/cta.go

```diff
@@ -12,12 +17,82 @@ func (cmd *generateNetpolCommand) generateNetpol() error {
 		return errors.Wrap(err, "Error synthesizing policies from folder")
 	}
 
+	var yamlPolicies []string
+	var mergedPolicy string
 	for _, netpol := range recommendedNetpols {
 		yamlPolicy, err := networkpolicy.KubernetesNetworkPolicyWrap{NetworkPolicy: netpol}.ToYaml()
 		if err != nil {
-			return errors.Wrap(err, "Error converting YAML into Network Policy")
+			return errors.Wrap(err, "Error converting Network Policy object to YAML")
+		}
+		yamlPolicies = append(yamlPolicies, yamlPolicy)
+		mergedPolicy = fmt.Sprintf("%s---\n%s", mergedPolicy, yamlPolicy)
+	}
+
+	if !cmd.mergePolicies && !cmd.splitPolicies {
+		cmd.printNetpolsToStdout(mergedPolicy)
+		return nil
+	}
+
+	if cmd.mergePolicies {
+		if err := cmd.saveNetpolsToMergedFile(mergedPolicy); err != nil {
+			return errors.Wrapf(err, "Error writing merged Network Policies to file")
+		}
+	}
+
+	if cmd.splitPolicies {
+		if err := cmd.saveNetpolsToFolder(recommendedNetpols); err != nil {
+			return errors.Wrapf(err, "Error writing split Network Policies to folder")
 		}
-		cmd.env.Logger().PrintfLn("---\n\n%s", yamlPolicy)
 	}
+
 	return nil
 }
+
+func (cmd *generateNetpolCommand) printNetpolsToStdout(combinedNetpols string) {
+	cmd.env.Logger().PrintfLn(combinedNetpols)
+}
+
+func (cmd *generateNetpolCommand) saveNetpolsToMergedFile(combinedNetpols string) error {
+	dirpath, filename := filepath.Split(cmd.outputFilePath)
+	if dirpath == "" {
+		dirpath = "./"
+	}
+	if filename == "" {
+		filename = "policies.yaml"
+	}
+
+	if err := writeFile(filename, dirpath, combinedNetpols); err != nil {
+		return errors.Wrapf(err, "Error writing policy to file")
+	}
+	return nil
+}
+
+func (cmd *generateNetpolCommand) saveNetpolsToFolder(recommendedNetpols []*v1.NetworkPolicy) error {
+	for _, netpol := range recommendedNetpols {
+		policyName := netpol.GetName()
+		if policyName == "" {
+			policyName = string(netpol.GetUID())
+		}
+		filename := policyName + ".yaml"
+
+		yamlPolicy, err := networkpolicy.KubernetesNetworkPolicyWrap{NetworkPolicy: netpol}.ToYaml()
+		if err != nil {
+			return errors.Wrap(err, "Error converting Network Policy object to YAML")
+		}
+
+		if err := writeFile(filename, cmd.outputFolderPath, yamlPolicy); err != nil {
+			return errors.Wrapf(err, "Error writing policy to file")
```

Comment:
Nit: the error message could read: _"Error writing policy to file: Error writing file: ..."_.
Looks like `writeFile` should not capitalize the first word in its error messages. And in general, we do not capitalize it.

Reply:
Thanks for the recommendation. I've updated the wrap messages.